fineract-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/Loan.java

```diff
@@ -4796,33 +4434,11 @@ public LoanTransaction getLastRepaymentOrDownPaymentTransaction() {
     }
 
     public LocalDate getLastUserTransactionForChargeCalc() {
-        LocalDate lastTransaction = getDisbursementDate();
-        if (this.repaymentScheduleDetail().isInterestRecalculationEnabled()) {
-            lastTransaction = getLastUserTransactionDate();
-        }
-        return lastTransaction;
+        return this.repaymentScheduleDetail().isInterestRecalculationEnabled() ? getLastUserTransactionDate() : getLastRepaymentDate();
```

Comment:
I know you will not like this one, but here the `getDisbursementDate()` need to be used and just leave comment, this needs to be reviewed!